x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecycleMetadata.java

```diff
@@ -132,11 +134,13 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.field(POLICIES_FIELD.getPreferredName(), this.snapshotConfigurations);
-        builder.field(OPERATION_MODE_FIELD.getPreferredName(), operationMode);
-        builder.field(STATS_FIELD.getPreferredName(), this.slmStats);
-        return builder;
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params ignored) {
+        return Iterators.single(((builder, params) -> {
+            builder.xContentValuesMap(POLICIES_FIELD.getPreferredName(), this.snapshotConfigurations);
```

Comment:
Why not chunk this? I mean ok it's probably not going to be very big in most cases but there is still potential for abuse here.

Reply:
I initially thought this was too small to makes sense. But now that you mentioned it ... makes sense actually, there's a potentially somewhat large exception in the metadata for each policy so it's worth it for the bad case :) -> added chunking here.